src/Context.php

```diff
@@ -119,6 +119,10 @@ public function addSymbol($name)
      */
     public function addVariable(Variable $variable)
     {
+        if (!is_string($variable->getName())) {
```

Comment:
But if I do that, variable variables (`${$foo} = 42;`) will not be compilable anymore by phpsa because they are currently represented using nested `Variable` instances :

```php
$variableVariable = new \PHPSA\Variable(
    new \PHPSA\Variable('foo', 'bar'),
    42
);
```

How do you want to handle this?